orttraining/orttraining/training_ops/cuda/activation/gelu_grad_impl_common.cuh

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "orttraining/training_ops/cpu/activation/gelu_computation_mode.h"
+
+namespace onnxruntime {
+namespace cuda {
+
+template <typename T>
+__device__ __inline__ T ComputeGeluGradScalar(T dY, T X, gelu_computation_mode::Default) {
```

Comment:
>gelu_computation_mode::Default [](start = 57, length = 30)

Just wondering why this becomes function parameter... 
I see it as template parameter in other files... 

I feel template parameter is more natural in this case

Reply:
function templates can't be partially specialized. it's possible to delegate to a partially specialized class template but i think this is less boilerplate.
btw this is a good read: [http://www.gotw.ca/publications/mill17.htm](http://www.gotw.ca/publications/mill17.htm)

---
In reply to: [441104853](https://github.com/microsoft/onnxruntime/pull/4146#discussion_r441104853) [](ancestors = 441104853)